### DISCUSSION TITLE
Is it possible to use CommonJs with AdonisJs version 6

### DISCUSSION THREAD

**Discussion by kaypee90** [adonisjs/core#4390] [Help] [answered]
I am trying to implement open telemetry (https://signoz.io/blog/nodejs-opensource-application-monitoring/) in Adonis Js version 6. This requires implementing a `tracing.js` which uses common. And this is throwing an exception `[Error [ERR_REQUIRE_ESM]: require() of ES Module tracing.js not supported`

Does this mean all packages that use only CommonJs are not supported in AdonisJs V6 or there is a way around this? Please help!

**Comment by RomainLanz** [comment]
Hey @kaypee90! 👋🏻 

The error states you are trying to load an ESM module in a CJS environment. So I am not sure what you have done. Could you share some code?

**Reply by kaypee90** [reply]
@RomainLanz  This is the content of the tracing.js file:
```
// tracing.js
'use strict'
const process = require('process');
const opentelemetry = require('@opentelemetry/sdk-node');
const { getNodeAutoInstrumentations } = require('@opentelemetry/auto-instrumentations-node');
const { OTLPTraceExporter } = require('@opentelemetry/exporter-trace-otlp-http');
const { Resource } = require('@opentelemetry/resources');
const { SemanticResourceAttributes } = require('@opentelemetry/semantic-conventions');

const exporterOptions = {
   url: 'http://localhost:4318/v1/traces'
   }
   
const traceExporter = new OTLPTraceExporter(exporterOptions);
const sdk = new opentelemetry.NodeSDK({
   traceExporter,
   instrumentations: [getNodeAutoInstrumentations()],
   resource: new Resource({
      [SemanticResourceAttributes.SERVICE_NAME]: 'node_app'
      })
});

// initialize the SDK and register with the OpenTelemetry API
// this enables the API to record telemetry

sdk.start()

// gracefully shut down the SDK on process exit
process.on('SIGTERM', () => {
   sdk.shutdown()
 .then(() => console.log('Tracing terminated'))
 .catch((error) => console.log('Error terminating tracing', error))
 .finally(() => process.exit(0));
 });
```


And I am starting the app using the command `node -r ./tracing.js ace serve --watch`

**Reply by RomainLanz** [reply, marked as answer]
Your file is coded in CJS; you have to name it `tracing.cjs` for it to be loaded as CJS.

**Comment by acidiney** [comment]
Hey @RomainLanz, I'm having a issue setting up this... If I use node --require tracing.cjs ace serve --watch, after adonis starting ... It stops sending events from app.

If I use as a preload on `adonisrc.ts` it actually send app events, but I receive a lot of warnings with others packages like knex that was not preload correctly.

Maybe `unstable_assembler` could help us, but, we don't have any beforeStarting hook, the only one we have, is actually  `onDevServerStarted`

Note: After build for production it actually works `node --require tracing.js bin/server.js`
